EIPS/eip-5007.md

```diff
@@ -0,0 +1,313 @@
+---
+eip: 5007
+title: TimeNFT, ERC-721 Time Extension
+description: Add start time and end time to ERC-721 tokens.
+author: Anders (@0xanders), Lance (@LanceSnow), Shrug <shrug@emojidao.org>
+discussions-to: https://ethereum-magicians.org/t/eip5007-erc721-time-extension/8924
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-04-13
+requires: 165, 721
+---
+
+## Abstract
+
+This standard is an extension of [ERC-721](./eip-721.md). It proposes some additional property( `startTime`, `endTime`,`originalTokenId`) to help with the on-chain time management.
+
+## Motivation
+
+Some NFTs have a defined usage period and cannot be used when they are not at a specific time. If you want to make NFT invalid when it is not in use period, or make NFT enabled at a specific time, while the NFT does not contain time information, you often need to actively submit the chain transaction, this process is both cumbersome and a waste of gas.
```

Comment:
Could you include an example of what kind of NFTs have a time component?

Reply:
> Could you include an example of what kind of NFTs have a time component?

1) [ENS](https://etherscan.io/address/0x57f1887a8bf19b14fc0df6fd9b2acc9af147ea85#readContract) :
nameExpires(id) of ENS   -->  endTime(id) of  EIP-5007 ;  
register time of ENS -->  startTime(id) of EIP-5007 ；


2) [doNFT](https://etherscan.io/address/0x8B81e64b6d4ebE2EC0f155df28c54aE2F20189DD#readContract) of Double Protocol: 
getDuration(id)[0] , start time of the lease   --> startTime(id) of EIP-5007 
getDuration(id)[1] , end time of the lease--> endTime(id) of EIP-5007

Reply:
Sorry, I meant in the EIP itself :rofl: